internal/convert/registryv1.go

```diff
@@ -1,34 +1,146 @@
 package convert
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
-	"hash"
+	"github.com/operator-framework/rukpak/internal/util"
 	"hash/fnv"
+	"io"
+	"io/fs"
+	"path/filepath"
 	"strings"
+	"testing/fstest"
+	"time"
 
-	"github.com/davecgh/go-spew/spew"
 	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	v2 "github.com/operator-framework/api/pkg/operators/v2"
+	"github.com/operator-framework/operator-registry/pkg/registry"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/apimachinery/pkg/util/sets"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 )
 
 type RegistryV1 struct {
-	CSV    v1alpha1.ClusterServiceVersion
-	CRDs   []apiextensionsv1.CustomResourceDefinition
-	Others []unstructured.Unstructured
+	PackageName string
+	CSV         v1alpha1.ClusterServiceVersion
+	CRDs        []apiextensionsv1.CustomResourceDefinition
+	Others      []unstructured.Unstructured
 }
 
 type Plain struct {
 	Objects []client.Object
 }
 
+func RegistryV1ToPlain(rv1 fs.FS) (fs.FS, error) {
+	reg := RegistryV1{}
+	fileData, err := fs.ReadFile(rv1, filepath.Join("metadata", "annotations.yaml"))
+	if err != nil {
+		return nil, err
+	}
+	annotationsFile := registry.AnnotationsFile{}
+	if err := yaml.Unmarshal(fileData, &annotationsFile); err != nil {
+		return nil, err
+	}
+	reg.PackageName = annotationsFile.Annotations.PackageName
```

Comment:
do we need to nil check this to avoid a panic here?

Reply:
actually, no we don't. the `Annotations` field of `annotationsFile` is not a pointer.